pokedex/src/App.js

```diff
@@ -1,109 +1,107 @@
+import { useOutletContext, useSearchParams } from "react-router-dom";
+import Pokedex from "./components/Pokedex";
+import Displaying from "./components/Displaying";
 
-const App = () => {
-  const [data, setData] = React.useState({});
-  const [meta, setMeta] = React.useState({});
-  const [links, setLinks] = React.useState({});
+const App = ({ context }) => {
   const [searchParams, setSearchParams] = useSearchParams();
   let name = searchParams.get("name");
   let page = searchParams.get("page");
-  const myInput = React.createRef();
+  const [pokedex, setPokedex] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [meta, setMeta] = useState({});
+  const myInput = createRef();
+  const { setReturnPage, returnPage } = useOutletContext();
+  
+
+  const changeReturnPage = (newPage) => {
+    if (name === "" && !isLoading) {
+      setReturnPage(newPage);
+    }
+  };
 
   const nextPage = () => {
     if (meta.current_page !== meta.last_page) {
-      setSearchParams({ page: meta.current_page + 1 });
+      setSearchParams({ name: name, page: meta.current_page + 1 });
+      changeReturnPage(meta.current_page+1)
     }
   };
 
   const previousPage = () => {
     if (meta.current_page !== 1) {
-      setSearchParams({ page: meta.current_page - 1 });
+      setSearchParams({ name: name, page: meta.current_page - 1 });
+      changeReturnPage(meta.current_page - 1);
     }
   };
-
-  const lastPage = () => {
-    setSearchParams({ page: meta.last_page });
-  };
-
-  const firstPage = () => {
-    setSearchParams({ page: 1 });
-  };
-
-  React.useEffect(() => {
-    if (name) {
-      getPokemon();
-    } else if (page) {
-      getPokemonPage();
-    } else {
-      fetch("https://intern-pokedex.myriadapps.com/api/v1/pokemon?")
-        .then((res) => res.json())
-        .then((json) => {
-          setData(json.data);
-          setLinks(json.links);
-          setMeta(json.meta);
-        });
-    }
-
-    async function getPokemonPage() {
-      await fetch("https://intern-pokedex.myriadapps.com/api/v1/pokemon?page=" + page)
-        .then((res) => res.json())
-        .then((json) => {
-          setData(json.data);
-          setLinks(json.links);
-          setMeta(json.meta);
-        });
-    }
-
-    async function getPokemon() {
-      await fetch("https://intern-pokedex.myriadapps.com/api/v1/pokemon?name=" + name)
-        .then((res) => res.json())
-        .then((json) => {
-          setData(json.data);
-          setLinks(json.links);
-          setMeta(json.meta);
-        });
+  
+  useEffect(() => {
+    setIsLoading(true);
+    setSearchParams({ name: name ? name : "", page: page ? page : 1 });
+    let url = "https://intern-pokedex.myriadapps.com/api/v1/pokemon?" + new URLSearchParams({ name, page });
+    if(page!==null){
+      fetch(url)
+      .then((res) => res.json())
+      .then((json) => {
+        setPokedex(json.data);
+        setMeta(json.meta);
+        setIsLoading(false);
+      });
```

Comment:
You can use object destructuring here and clean it up a little bit.

`
      .then(({ data, meta }) => {
        setPokedex(data);
        setMeta(meta);
      });
`